x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/EvalMapper.java

```diff
@@ -29,29 +37,14 @@
 import org.elasticsearch.xpack.ql.expression.predicate.logical.Not;
 import org.elasticsearch.xpack.ql.expression.predicate.nulls.IsNotNull;
 import org.elasticsearch.xpack.ql.expression.predicate.nulls.IsNull;
-import org.elasticsearch.xpack.ql.util.ReflectionUtils;
 
 import java.util.List;
 import java.util.function.IntFunction;
 import java.util.function.Supplier;
 
 public final class EvalMapper {
-    abstract static class ExpressionMapper<E extends Expression> {
-        private final Class<E> typeToken;
-
-        protected ExpressionMapper() {
-            typeToken = ReflectionUtils.detectSuperTypeForRuleLike(getClass());
-        }
-
-        protected abstract Supplier<ExpressionEvaluator> map(E expression, Layout layout);
-    }
 
     private static final List<ExpressionMapper<?>> MAPPERS = List.of(
-        ArithmeticMapper.ADD,
-        ArithmeticMapper.DIV,
-        ArithmeticMapper.MOD,
-        ArithmeticMapper.MUL,
-        ArithmeticMapper.SUB,
         ComparisonMapper.EQUALS,
         ComparisonMapper.NOT_EQUALS,
         ComparisonMapper.GREATER_THAN,
```

Comment:
question: since `Neg` now implements `EvaluatorMapper`, has `NegMapper.NEG_MAPPER` in the list below become obsolete? (If so, the `NegMapper` class as a whole may be obsolete now.)

Reply:
This has been fixed but I forgot to push my latest commits...